packages/js/src/utils/scopes.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * @param originalScopes
+ * @return scopes should contain all reserved scopes ( Logto requires `openid` and `offline_access` )
+ */
+export const withReservedScopes = (originalScopes?: string[]): string => {
+  const uniqueScopes = new Set(['openid', 'offline_access', ...(originalScopes ?? [])]);
+
+  return Array.from(uniqueScopes).slice().sort().join(' ');
```

Comment:
What does the `slice()` do? Shallow copy the array?
I think `Array.from()` already does the shallow copy, so it's not necessary to `slice()` again

Reply:
it's lint requirement to prevent unexpected inline array sorting

Reply:
I see. It is to avoid direct mutation

Reply:
`sort` is unnecessary. will remove `sort`